Library/Homebrew/formula.rb

```diff
@@ -923,9 +957,23 @@ def exec_cmd(cmd, args, out, logfn)
     exit! 1 # never gets here unless exec threw or failed
   end
 
+  def source_dir
+    if build.dsym?
+      prefix + "Source"
+    else
+      nil
+    end
+  end
```

Comment:
the way -g  and  dsymutil work on OS X is almost all of the debug information is retained in the object files, and not copied to the final executable when it is linked.   The executable only has links to the object files.   So in order to collect dsyms we need to run dsymutil before all the object files have been cleaned up.   

Also, we want to retain the source files so the debugger will work, but we don't want to retain all those object files, so I made the build directory be $KEG/Source instead of a temp dir if --dsym is selected, and it's up to the formula to make sure that it doesn't leave a bunch of object files sitting there.